local.go

```diff
@@ -346,7 +389,8 @@ func update(location *Location, optLoc *Location, stats *Stats, iterType Iterati
 	if iterType == MajorIteration {
 		stats.MajorIterations++
 	}
-	if location.F < optLoc.F {
+	// TODO: Update optLoc only at MajorIterations?
```

Comment:
I don't think we want to do this. For example, a linesearch may find a better function value but not have sufficient gradient decrease to meet Armijio. It wants to continue the linesearch, but the function evaluation limit is reached. We want to return the best point found in this case I would imagine.